### PR TITLE
Fix/hosting/private sql

### PR DESCRIPTION
**Pull request by FredericEspiau**
DTRSD-8901

# Validation
This has already been validated by three of my peers, but I've rebased in the meanwhile

### CODE REVIEW

File: packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js [hunk 1]
```diff
@@ -393,35 +394,22 @@ angular.module('services').service(
...
-    getPrivateDatabaseCapabilities(serviceName) {
+    getHasPrivateSqlToActivate(serviceName) {
       return this.OvhHttp.get(
-        `/hosting/web/${serviceName}/privateDatabaseCreationCapabilities`,
+        `/order/cartServiceOption/webHosting/${serviceName}`,
         {
           rootPath: 'apiv6',
         },
-      );
+      ).then((options) => some(options, { planCode: 'private-sql-512' }));
```
Comment (varun257) [inline packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js]: `planCode` could come from a constant

File: packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js [hunk 2]
```diff
@@ -393,35 +394,22 @@ angular.module('services').service(
...
-        {
-          rootPath: 'apiv6',
-          data: {
-            ram,
-            version,
-          },
-        },
-      );
+    getPrivateDatabaseIds(serviceName) {
+      return this.OvhHttp.get(`/hosting/web/${serviceName}/privateDatabases`, {
```
Comment (radireddy) [inline packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js]: This should come from ovh-api-services?

File: packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js [hunk 3]
```diff
@@ -393,35 +394,22 @@ angular.module('services').service(
...
 
-    /**
-     * Get private database capabilities
-     * @param {string} serviceName
-     */
-    getPrivateDatabaseCapabilities(serviceName) {
+    getHasPrivateSqlToActivate(serviceName) {
       return this.OvhHttp.get(
-        `/hosting/web/${serviceName}/privateDatabaseCreationCapabilities`,
+        `/order/cartServiceOption/webHosting/${serviceName}`,
```
Comment (radireddy) [inline packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js]: same again should come from ovh-api-services

File: packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js (hunk 1 above)
Comment (FredericEspiau) [reply packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js]: Done !

File: packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js (hunk 2 above)
Comment (cbourgois) [reply packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js]: Yes, it (sh|c)ould, but as it is a fix refactor thing is not mandatory.

File: packages/manager/apps/web/client/app/private-database/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -417,65 +417,6 @@
   "privateDatabase_metrics_time_range_DAY": "Dernières 24 heures",
   "privateDatabase_metrics_time_range_WEEK": "Dernière semaine",
   "privateDatabase_metrics_time_range_MONTH": "Dernier mois",
-  "privateDatabase_order_sql_main_title": "Commander une base de données SQL",
```
Comment (cbourgois) [inline packages/manager/apps/web/client/app/private-database/translations/Messages_fr_FR.json]: We could clean the translations targets files when merged.

File: packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js (hunk 2 above)
Comment (FredericEspiau) [reply packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js]: Also it should be merged ASAP and I don't want to have to wait to prod `ovh-api-services` 😅

File: packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js (hunk 3 above)
Comment (FredericEspiau) [reply packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js]: It should be merged ASAP and I don't want to have to wait to prod `ovh-api-services` 😅

File: packages/manager/apps/web/client/app/private-database/translations/Messages_fr_FR.json (hunk 4 above)
Comment (FredericEspiau) [reply packages/manager/apps/web/client/app/private-database/translations/Messages_fr_FR.json]: What do you mean ? Remove them from this PR and do it later in another PR on `develop` ? I considered it and didn't know what was the best procedure

Comment (antleblanc) [reply packages/manager/apps/web/client/app/private-database/translations/Messages_fr_FR.json]: We will automatize this process. It's related to #2284.

For now we can keep this as is.

**Comment by antleblanc** [comment]
Regarding the commit: 01b1d3a1f5951012b07d534e67371d157d361c70

Both following modules are updated:
> server-sidebar: packages/manager/modules/server-sidebar/src/order.constants.js
> core: packages/manager/modules/core/src/manager-core.constants.js

They deserve a patch. So it could be splitted into multiple commits type like a `fix` for instance.

**Comment by FredericEspiau** [comment]
@antleblanc done

File: packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js [hunk 5]
```diff
@@ -393,35 +396,22 @@ angular.module('services').service(
...
+        rootPath: 'apiv6',
+      });
     }
 
-    /**
-     * Get private database capabilities
-     * @param {string} serviceName
-     */
-    getPrivateDatabaseCapabilities(serviceName) {
+    getHasPrivateSqlToActivate(serviceName) {
```
Comment (frenautvh) [inline packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js]: getHas is not really conventional nor shakespearian ^^ "has" seems enough

File: packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 6]
```diff
@@ -450,66 +450,65 @@
...
+                            ></p>
+
                             <p
-                                data-ng-if="hosting.sqlPriveInfo.nbDataBaseInclude > hosting.sqlPriveInfo.nbDataBaseActive"
+                                data-ng-if="hosting.sqlPriveInfo.hasPrivateDatabaseToActivate"
                                 data-translate="hosting_dashboard_service_privatesql_active_remaining"
-                                data-translate-values="{
-                                   count: hosting.sqlPriveInfo.nbDataBaseInclude - hosting.sqlPriveInfo.nbDataBaseActive,
-                               }"
+                                data-translate-values="{ count: 1 }"
```
Comment (frenautvh) [inline packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: Are you sure it's always one? Then maybe the sentence should be rewritten to something like : 'Activer votre base de donnée privée' ?

File: packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 7]
```diff
@@ -450,66 +450,65 @@
...
+                            <div
+                                data-ng-if="!hosting.sqlPriveInfo.hasPrivateDatabaseToActivate"
+                                data-ng-repeat="privateDb in privateDatabases track by privateDb.name"
+                            >
+                                <oui-action-menu-item data-disabled>
+                                    <span data-ng-bind="privateDb.name"></span>
+                                </oui-action-menu-item>
+                                <oui-action-menu-item
+                                    data-ng-if="privateDb.isAdmin"
+                                    data-href="#/configuration/private_database/{{ privateDb.name }}"
```
Comment (frenautvh) [inline packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: if possible, use ui-router states, in the routing

File: packages/manager/apps/web/client/app/hosting/hosting.constant.js [hunk 8]
```diff
@@ -21,4 +21,17 @@ angular
     DELETING: 'deleting',
     REGENERATING: 'regenerating',
     UPDATING: 'updating',
-  });
+  })
+  .constant('HOSTING_UPGRADES', [
```
Comment (frenautvh) [inline packages/manager/apps/web/client/app/hosting/hosting.constant.js]: can't you fetch this enum from the API ?

File: packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js (hunk 5 above)
Comment (FredericEspiau) [reply packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js]: `has` doesn't convey the fact that it's a `Promise` 😉  `getHas` means `it's a Promise<boolean>`

File: packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.html (hunk 6 above)
Comment (FredericEspiau) [reply packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: It's a DTrun, I won't change the translations, but I really want to 😢

File: packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.html (hunk 7 above)
Comment (FredericEspiau) [reply packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: Same story, it's a DTrun, it has been running for a whole week, I already did too much 😅

File: packages/manager/apps/web/client/app/hosting/hosting.constant.js (hunk 8 above)
Comment (FredericEspiau) [reply packages/manager/apps/web/client/app/hosting/hosting.constant.js]: Good point !

**Comment by FredericEspiau** [comment]
@frenautvh 

I won't do your remarks as I want this to be prodded ASAP, as they are really good I've created an improvement JIRA that I intend to include in the next sprint: MANAGER-4581

**Pull request merged**